apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/DispatcherServletRenderInstrumentation.java

```diff
@@ -0,0 +1,93 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.spring.webmvc;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.impl.transaction.AbstractSpan;
+import co.elastic.apm.agent.impl.transaction.Span;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.annotation.Nullable;
+
+import static net.bytebuddy.matcher.ElementMatchers.declaresMethod;
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.isInterface;
+import static net.bytebuddy.matcher.ElementMatchers.nameStartsWith;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.not;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class DispatcherServletRenderInstrumentation extends ElasticApmInstrumentation {
+
+    private static final String SPAN_TYPE_DISPATCHER_SERVLET_RENDER = "template.dispatcher-servlet.render";
+    private static final String RENDER_METHOD = "Render";
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    private static void beforeExecute(@Advice.Argument(0) ModelAndView modelAndView,
+                                      @Advice.Local("span") Span span) {
+        if (tracer == null || tracer.activeSpan() == null)
+            return;
+        final AbstractSpan<?> parent = tracer.activeSpan();
+        span = parent
+            .createSpan()
+            .withType(SPAN_TYPE_DISPATCHER_SERVLET_RENDER)
+            .appendToName(RENDER_METHOD).appendToName(" ").appendToName(modelAndView.getViewName() != null ? modelAndView.getViewName() : "")
```

Comment:
If view name is not available, maybe a better name would be `DispatcherServlet#render` to be more Spring MVC indicative

Reply:
i can't reproduce the case when ModelAndView#getViewName returns null, in DispatcherServlet there is a check, if View is null, viewName is set from the request URI:
```
	protected String getDefaultViewName(HttpServletRequest request) throws Exception {
		return (this.viewNameTranslator != null ? this.viewNameTranslator.getViewName(request) : null);
	}
```

Reply:
This can also return null..

Reply:
Exactly what I meant, thanks!
Only I would go with
```
    private static final String RENDER_METHOD = "Render";
    private static final String DISPATCHER_SERVLET_RENDER_METHOD = "DispatcherServlet#render";

    @Advice.OnMethodEnter(suppress = Throwable.class)
    private static void beforeExecute(@Advice.Argument(0) ModelAndView modelAndView,
                                      @Advice.Local("span") Span span) {
        if (tracer == null || tracer.activeSpan() == null || isAlreadyMonitored(tracer.activeSpan()))
            return;
        final AbstractSpan<?> parent = tracer.activeSpan();
        
        span = parent.createSpan().withType(SPAN_TYPE_DISPATCHER_SERVLET_RENDER);
        if (modelAndView == null || modelAndView.getViewName() == null)
            span.appendToName(DISPATCHER_SERVLET_RENDER_METHOD);
        else
            span.appendToName(RENDER_METHOD).appendToName(" ").appendToName(modelAndView.getViewName());

        span.activate();
    }
```
Just because I am allergic to code duplications 😊

Reply:
Maybe use `DISPATCHER_SERVLET_RENDER_METHOD` in both cases and add `modelAndView.getViewName()` if available to be consistent.

Also, in the actual code, please use braches for the if/else statements.